pkg/modules/spanpercentile/implspanpercentile/translator_cte_builder_test.go

```diff
@@ -0,0 +1,339 @@
+package implspanpercentile
+
+import (
+	"fmt"
+	qbtypes "github.com/SigNoz/signoz/pkg/types/querybuildertypes/querybuildertypesv5"
+	"github.com/SigNoz/signoz/pkg/types/spanpercentiletypes"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestSpanPercentileCTEBuilder(t *testing.T) {
+	cases := []struct {
+		name        string
+		request     *spanpercentiletypes.SpanPercentileRequest
+		expected    qbtypes.Statement
+		expectedErr error
+	}{
+		{
+			name: "basic span percentile query without filter",
+			request: &spanpercentiletypes.SpanPercentileRequest{
+				SpanID: "span123456789abcdef",
+				Start:  1640995200000,
+				End:    1640995800000,
+			},
+			expected: qbtypes.Statement{
+				Query: "WITH base_spans AS (SELECT *, resource_string_service$$name AS `service.name` FROM signoz_traces.distributed_signoz_index_v3 WHERE timestamp >= '1640995200000000000' AND timestamp < '1640995800000000000' AND ts_bucket_start >= 1640993400 AND ts_bucket_start <= 1640995800), target_span AS (SELECT duration_nano, name, `service.name` as service_name, resources_string['deployment.environment'] as deployment_environment FROM base_spans WHERE span_id = 'span123456789abcdef' LIMIT 1) SELECT 'span123456789abcdef' as span_id, t.duration_nano, t.duration_nano as duration_ms, t.name as span_name, t.service_name, t.deployment_environment, round((sum(multiIf(s.duration_nano < t.duration_nano, 1, 0)) * 100.0) / count(*), 2) as percentile_position, quantile(0.50)(s.duration_nano) as p50_duration_ms, quantile(0.90)(s.duration_nano) as p90_duration_ms, quantile(0.99)(s.duration_nano) as p99_duration_ms, count(*) as total_spans_in_group FROM target_span t CROSS JOIN base_spans s WHERE s.name = t.name AND s.`service.name` = t.service_name AND s.resources_string['deployment.environment'] = t.deployment_environment GROUP BY t.duration_nano, t.name, t.service_name, t.deployment_environment",
+				Args:  nil,
+			},
+			expectedErr: nil,
+		},
+		{
+			name: "span percentile query with resource filter",
+			request: &spanpercentiletypes.SpanPercentileRequest{
+				SpanID: "span987654321fedcba",
+				Start:  1640995200000,
+				End:    1640995800000,
+				Filter: &qbtypes.Filter{
+					Expression: "simpleJSONExtractString(labels, 'service.name') = 'frontend' AND labels LIKE '%service.name%' AND labels LIKE '%service.name\":\"frontend%'",
```

Comment:
we cannot write these kind of queries from the UI

we can only write like this
`service.name = 'frontend'`